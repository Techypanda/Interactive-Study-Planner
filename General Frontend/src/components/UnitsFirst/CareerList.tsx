import { Box, Paper, Typography } from "@material-ui/core";
import { Work } from "@material-ui/icons";
import styled from "styled-components";
import { useRemainingHeight } from "../../api/hooks";
import { CareerListEntryProps, CareerListSPAProps } from "../../types";

function CareerListEntry(props: CareerListEntryProps) {
  return (
    <Paper className={props.className}>
      <Box display="flex" className={props.className} py={2}>
        <Box marginLeft={2}>
          <Typography variant="subtitle1" className="careername">{props.title}</Typography>
        </Box>
        <Box flexGrow={1} />
        <Box marginRight={2}>
          <Work className="careericon" />
        </Box>
      </Box>
    </Paper>
  )
}

function CareerList(props: CareerListSPAProps) {
  const height = useRemainingHeight();
  return (
    <div className={props.className}>
      <Box className="careerList" minHeight={height}>
        <Box pt={2}>
          <Typography variant="h5">Available Careers</Typography>
          {props.careers.map((career) => <CareerListEntry className="stylethetext clickable" title={career.Name} key={career.CareerId} />)}
        </Box>
      </Box>
    </div>
  )
}
export default styled(CareerList)`
.careerList {
  border-left: 2px solid #cc9900;
}
.careername {
  display: inline-block;
}
.careericon {
  vertical-align: bottom;
}
.stylethetext {
  text-transform: capitalize;
}
.clickable {
  cursor: pointer;
}
`;