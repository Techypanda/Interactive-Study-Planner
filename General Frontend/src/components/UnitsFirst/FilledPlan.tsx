import { Box, Button, Container, Typography } from "@material-ui/core";
import { useHistory } from "react-router";
import styled from "styled-components";
import { DefaultProps } from "../../types";

function FilledPlan(props: DefaultProps) {
  const history = useHistory()
  return (
    <Container className={props.className}>
      <Box mt={8}>
        <Box mb={2}>
          <Typography variant="h5">Your Plan Is Now Full</Typography>
        </Box>
        <Typography variant="subtitle1">You can view your available career paths on the list to the right of the screen, if you want more information you can click on the career to open a career information view.<br />You can also click continue to move onto planning your study plan.</Typography>
        <Box mt={3}>
          <Box mr={2} display="inline">
            <Button className="curtinBtn" variant="contained" onClick={() => history.push('/')}>Return To Main Menu</Button>
          </Box>
          <Box ml={2} display="inline">
            <Button className="curtinBtn" variant="contained" onClick={() => history.push('/timetable')}>Continue To Study Plan</Button>
          </Box>
        </Box>
      </Box>
    </Container>
  )
}
export default styled(FilledPlan)`
.curtinBtn {
  background-color: #f3c32d !important;
}
`;