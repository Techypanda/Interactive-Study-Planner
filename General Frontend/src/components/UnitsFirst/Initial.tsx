import { Box, Button, Typography, useMediaQuery } from "@material-ui/core";
import { NavigateBefore, NavigateNext } from "@material-ui/icons";
import { useState } from "react";
import styled from "styled-components";
import { InitialCareerSPAProps } from "../../types";
import OptionCard from "./OptionCard";

// Viewport height - element.offset.top - desired bottom margin
function Initial(props: InitialCareerSPAProps) {
  const switchToOneCard = useMediaQuery('(max-width: 1500px)');
  const [currentSelection, setCurrentSelection] = useState(0);
  const navPrevious = () => {
    const prevIDX = currentSelection - 1 < 0 ? props.majors.length - 1 : currentSelection - 1;
    setCurrentSelection(prevIDX);
  }
  const navNext = () => {
    const nextIDX = currentSelection + 1 >= props.majors.length ? 0 : currentSelection + 1;
    setCurrentSelection(nextIDX);
  }
  const selectMajor = () => {
    props.selectMajor(props.majors[currentSelection])
  }
  return (
    <Box className={props.className} minHeight="100%" minWidth="100%" display="flex" alignItems="center" justifyContent="center">
      <Box pt={2}>
        <Typography variant="h4">Please Select Your First Major</Typography>
        <Box display="flex" justifyContent="center" mt={8}> {/* TODO: Make these slide around like pokemon starter selector */}
          {(props.majors.length >= 3 && !switchToOneCard) &&
            <OptionCard
              className="inner card-outer-left"
              style={{ transform: "translate(50px, -50px)" }}
              title="I'm a really cool blurred 3rd major"
              description="yep im bluured"
              type="Major"
            />
          }
          <OptionCard
            className="inner card-center"
            title={props.majors[currentSelection].Name}
            description={props.majors[currentSelection].Description}
            type="Major"
            onClick={() => selectMajor()}
          />
          {(props.majors.length >= 2 && !switchToOneCard) &&
            <OptionCard
              className="inner card-outer-right"
              style={{ transform: "translate(-50px, -50px)" }}
              title="I'm a really cool blurred 2nd major"
              description="yep im bluured"
              type="Major"
            />
          }
        </Box>
        <Box display="flex" justifyContent="center" mt={2}>
          <Button variant="contained" color="primary" onClick={() => selectMajor()}>Select</Button>
        </Box>
        <Box display="flex" alignItems="center" justifyContent="center" mt={4}>
          <Box mr={2}>
            <NavigateBefore className="navigationIcon" onClick={() => navPrevious()} />
          </Box>
          {props.majors.map((_, idx) => <Box mx={1} key={idx} className={`navIndicator ${idx === currentSelection ? 'activewoopog' : ''}`} display="inline-block" height={15} width={15} borderRadius="100%" />)}
          <Box ml={2}>
            <NavigateNext className="navigationIcon" onClick={() => navNext()} />
          </Box>
        </Box>
      </Box>
    </Box>
  )
}
export default styled(Initial)`
.inner {
  position: relative;
}
.navigationIcon {
  background-color: #cc9900 !important;
  color: #FFF !important;
  font-size: 50px !important;
  border-radius: 100%;
  padding: 5px;
  cursor: pointer;
}
.navIndicator {
  background-color: #d9d9d9;
  border: 2px solid #777777;
}
.activewoopog {
  background-color: #777777;
}
.card-outer-right {
  z-index: -1;
  filter: blur(4px);
}
.card.center {
  z-index: 0;
}
.card-outer-left {
  z-index: -1;
  filter: blur(4px);
}
`;