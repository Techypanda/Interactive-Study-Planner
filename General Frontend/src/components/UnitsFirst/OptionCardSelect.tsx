import { Card, CardHeader, CardMedia, CardContent, Typography } from "@material-ui/core";
import styled from "styled-components";
import { OptionCardProps } from "../../types";

function OptionCardSelect(props: OptionCardProps) {
  const styles = {
    height: "100%",
    ...props.style
  }
  return (
    <div className={props.className} style={styles} onClick={props.onClick}>
      <Card className="cardselection clickable">
        <CardHeader title={props.title} subheader={props.type} className="capit" />
        <CardMedia
          component="img"
          height="194"
          image="https://images.squarespace-cdn.com/content/v1/5e408d9d0b0aed2f2768b9aa/1585703579456-3UVMWL3XQM0XIIUHHQS8/faq-temp-img.png"
          alt="Major Card"
        />
        <CardContent>
          <Typography className="maxHeight">{props.description}</Typography>
        </CardContent>
      </Card>
    </div>
  )
}
export default styled(OptionCardSelect)`
.clickable {
  cursor: pointer;
}
.cardselection {
  width: 350px;
  height: 380px;
}
.maxHeight {
  max-height: 10px;
}
.capit {
  text-transform: capitalize;
}
`;