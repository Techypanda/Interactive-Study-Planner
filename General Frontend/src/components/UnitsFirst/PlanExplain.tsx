import { Box, Tooltip, Typography } from "@material-ui/core";
import styled from "styled-components";
import { PlanExplainProps } from "../../types";

function PlanExplain(props: PlanExplainProps) {
  return (
    <div className={props.className}>
      <Box display="flex" py={1} px={2} className="optionSelect">
        <Box pr={2}>
          <Typography variant="caption" display="inline" className="optionSelectTitle">{props.title}</Typography>
        </Box>
        <Box flexGrow={1} />
        <Tooltip title={props.explaination}>
          <Typography className="question">?</Typography>
        </Tooltip>
      </Box>
    </div>
  )
}
export default styled(PlanExplain)`
.optionSelect {
  border: 2px solid #cc9900;
}
.question {
  background-color: #1473ab;
  height: 25px;
  width: 25px;
  border-radius: 100%;
  border: 2px solid #cc9900;
  color: #FFF !important;
  cursor: pointer;
}
.optionSelectTitle {
  line-height: 29px;
}
`;