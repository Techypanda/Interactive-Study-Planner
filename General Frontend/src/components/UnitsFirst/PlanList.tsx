import { Box, MenuItem, Select, Typography } from "@material-ui/core";
import Clear from "@material-ui/icons/Clear";
import styled from "styled-components";
import { useRemainingHeight } from "../../api/hooks";
import { PlanProps } from "../../types";
function titleCase(str: String) { // https://stackoverflow.com/questions/32589197/how-can-i-capitalize-the-first-letter-of-each-word-in-a-string-using-javascript
  var splitStr = str.toLowerCase().split(' ');
  for (var i = 0; i < splitStr.length; i++) {
    splitStr[i] = splitStr[i].charAt(0).toUpperCase() + splitStr[i].substring(1);
  }
  return splitStr.join(' ');
}
function PlanList(props: PlanProps) {
  const menuVal = props.mainMajor?.Name;
  const height = useRemainingHeight();
  return (
    <div className={`${props.className} fh mh`}>
      <Box className="planList" minHeight={height} height="100%">
        <Box pt={2}>
          <Typography variant="h5">Current Plan</Typography>
          {menuVal ?
            <Select
              variant="outlined"
              fullWidth
              value={menuVal}
              renderValue={(_: any) => { return `Main Major: ${titleCase(menuVal)}` }}
              onChange={(event) => {
                props.updateMainMajor(props.majors[event.target.value as number])
              }}
            >
              {props.majors.map((major, idx) => <MenuItem key={major.MajorCode} value={idx}>
                {major.Name}
              </MenuItem>)}
            </Select> : <></>
          }
          {props.plan.doubleMajor &&
            <Box p={2} border="1px solid #000" display="flex">
              <Typography className="clipIfTooLong">DMajor: {props.plan.doubleMajor.Name}</Typography>
              <Box flexGrow={1} />
              <Clear className="clearColor" onClick={() => props.removeFromPlan(props.plan.doubleMajor!)} />
            </Box> 
          }
          {props.plan.optionalUnits?.map((unit) => 
            <Box p={2} border="1px solid #000" display="flex">
              <Typography className="clipIfTooLong">OUnit: {unit.Name}</Typography>
              <Box flexGrow={1} />
              <Clear className="clearColor" onClick={() => props.removeFromPlan(unit)} />
            </Box>
          )}
          {props.plan.specializations?.map((spec) => 
            <Box p={2} border="1px solid #000" display="flex">
              <Typography className="clipIfTooLong">{spec.Internal ? "ISpec" : "ESpec"}: {spec.Name}</Typography>
              <Box flexGrow={1} />
              <Clear className="clearColor" onClick={() => props.removeFromPlan(spec)} />
            </Box>
          )}
        </Box>
      </Box>
    </div>
  )
}
export default styled(PlanList)`
.planList {
  border-right: 2px solid #cc9900;
}
.clipIfTooLong {
  max-width: 80%;
  overflow: hidden;
  white-space: nowrap;
}
.clearColor {
  color: #b42121;
  cursor: pointer;
}
`;