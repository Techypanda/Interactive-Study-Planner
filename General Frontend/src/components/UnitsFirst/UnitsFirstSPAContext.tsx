import { Box, Grid } from "@material-ui/core";
import { useEffect, useState } from "react";
import styled from "styled-components";
import { Career, Major, Specialization, Unit, UnitFirstSPAContextProps } from "../../types";
import CareerList from "./CareerList";
import Initial from "./Initial";
import PlanList from "./PlanList";
import Workspace from "./Workspace";

enum UNITSFIRSTMODES {
  initial,
  workspace
}

function UnitsFirstSPAContext(props: UnitFirstSPAContextProps) {
  const [stage, setStage] = useState<UNITSFIRSTMODES>(UNITSFIRSTMODES.initial)
  const [careers, setCareers] = useState<Array<Career>>(props.careers);
  const [units, setUnits] = useState<Array<Unit>>(props.units);
  const [majors, setMajors] = useState<Array<Major>>(props.majors);
  const [specs, setSpecs] = useState<Array<Specialization>>(props.specs);
  const [mainMajor, setMMajor] = useState<Major>();
  function setMainMajor(major: Major) {
    setMMajor(major);
    setStage(UNITSFIRSTMODES.workspace);
  }


  return (
    <Box>
      <Grid container className="sameheight"> { /* this isnt going to work on mobile :/ */}
        <Grid item xs={2} className="sameheight">
          <PlanList
            mainMajor={mainMajor}
            majors={majors}
            updateMainMajor={(m: Major) => setMMajor(m)}
          />
        </Grid>
        <Grid item xs={8} className="sameheight">
          {stage === UNITSFIRSTMODES.initial ? <Initial majors={majors} selectMajor={setMainMajor} />
            : stage === UNITSFIRSTMODES.workspace ? <Workspace majors={majors} units={units} specs={specs} />
            : <h1>Unknown Stage</h1>
          }
        </Grid>
        <Grid item xs={2} className="sameheight">
          <CareerList careers={careers} />
        </Grid>
      </Grid>
    </Box >
  )
}
export default styled(UnitsFirstSPAContext)`
      `;