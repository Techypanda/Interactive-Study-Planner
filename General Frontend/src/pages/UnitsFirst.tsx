import { Box } from "@material-ui/core";
import styled from "styled-components";
import { useSpecializations } from "../api/hooks";
import { useCareers, useMajors, useUnits } from "../components/shared/hooks";
import { Career, DefaultProps } from "../types";
import { BounceLoader } from "react-spinners";
import UnitsFirstSPAContext from "../components/UnitsFirst/UnitsFirstSPAContext";

function UnitsFirst(props: DefaultProps) { // Bootstrap component for SPA Context
  const careers = useCareers();
  const majors = useMajors();
  const units = useUnits();
  const specs = useSpecializations();
  return (
    <Box className={props.className}> {/* Horrible Javascript Abuse TODO: Rework this */}
      {careers.isLoading || majors.isLoading || units.isLoading || specs.isLoading
        ? <Box my={2}><BounceLoader color="#1473AB" loading={true} size={150} /></Box>
        :
        <>
          {careers.isError || majors.isError || units.isError || specs.isError
            ? <h1>Error Has Occured</h1>
            : <UnitsFirstSPAContext careers={careers.data?.data!} majors={majors.data?.data!.sort((a, b) => a.Name.localeCompare(b.Name))!} units={units.data?.data!.sort((a, b) => a.Name.localeCompare(b.Name))!} specs={specs.data?.data!.sort((a, b) => a.Name.localeCompare(b.Name))!} />
          }
        </>
      }
    </Box>
  )
}
export default styled(UnitsFirst)`
`