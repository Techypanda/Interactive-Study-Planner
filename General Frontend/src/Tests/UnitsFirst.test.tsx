import React from 'react';
import { render } from '@testing-library/react';
import UnitsFirst from '../pages/UnitsFirst';
import Initial from '../components/UnitsFirst/Initial';
import { BrowserRouter as Router } from "react-router-dom";
import { QueryClient, QueryClientProvider } from 'react-query';
import { Major } from '../types';

describe('Units First works when', () => {
    const client = new QueryClient();
    it('Renders without crashing', () => {
        render(<QueryClientProvider client={client}>
            <Router>
                <UnitsFirst />
            </Router>
        </QueryClientProvider>);
    })
    it('Renders and matches snapshot', () => {
        const x = render(<QueryClientProvider client={client}>
            <Router>
                <UnitsFirst />
            </Router>
        </QueryClientProvider>);
        expect(x).toMatchSnapshot()
    })
})

describe('Initial bottom up component works when', () => {
    const client = new QueryClient();
    const mockMajors: Array<Major> = [{
        MajorCode: "",
        Name: "string",
        Description: "string",
        Credits: 2,
        Units: ["", ""],
        UnitAntiReqs: [[""], ["a"]],
        SpecAntiReqs: [[""], ["a"]]
    }]
    const mockSelectMajor = (m: Major) => { }
    it('Renders without crashing', () => {
        render(<QueryClientProvider client={client}>
            <Initial majors={mockMajors} selectMajor={mockSelectMajor} />
        </QueryClientProvider>)
    })
    it('Renders and matches snapshot', () => {
        const x = render(<QueryClientProvider client={client}>
            <Initial majors={mockMajors} selectMajor={mockSelectMajor} />
        </QueryClientProvider>);
        expect(x).toMatchSnapshot();
    })
})
