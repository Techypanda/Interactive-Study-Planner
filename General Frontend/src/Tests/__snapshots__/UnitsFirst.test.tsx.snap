// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Initial bottom up component works when Renders and matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiBox-root MuiBox-root-13 sc-eCstlR bBUCGc"
      >
        <div
          class="MuiBox-root MuiBox-root-14"
        >
          <h4
            class="MuiTypography-root MuiTypography-h4"
          >
            Please Select Your First Major
          </h4>
          <div
            class="MuiBox-root MuiBox-root-15"
          >
             
            <div
              class="sc-hKgJUU bfONkU inner card-center"
              style="height: 100%;"
            >
              <div
                class="MuiPaper-root MuiCard-root cardselection clickable MuiPaper-elevation1 MuiPaper-rounded"
              >
                <div
                  class="MuiCardHeader-root capit"
                >
                  <div
                    class="MuiCardHeader-content"
                  >
                    <span
                      class="MuiTypography-root MuiCardHeader-title MuiTypography-h5 MuiTypography-displayBlock"
                    >
                      string
                    </span>
                    <span
                      class="MuiTypography-root MuiCardHeader-subheader MuiTypography-body1 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
                    >
                      Major
                    </span>
                  </div>
                </div>
                <img
                  alt="Major Card"
                  class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img"
                  height="194"
                  src="https://images.squarespace-cdn.com/content/v1/5e408d9d0b0aed2f2768b9aa/1585703579456-3UVMWL3XQM0XIIUHHQS8/faq-temp-img.png"
                />
                <div
                  class="MuiCardContent-root"
                >
                  <p
                    class="MuiTypography-root maxHeight MuiTypography-body1"
                  >
                    string
                  </p>
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiBox-root MuiBox-root-16"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                Select
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
          <div
            class="MuiBox-root MuiBox-root-17"
          >
            <div
              class="MuiBox-root MuiBox-root-18"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root navigationIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
                />
              </svg>
            </div>
            <div
              class="MuiBox-root MuiBox-root-19 navIndicator activewoopog"
            />
            <div
              class="MuiBox-root MuiBox-root-20"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root navigationIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiBox-root MuiBox-root-13 sc-eCstlR bBUCGc"
    >
      <div
        class="MuiBox-root MuiBox-root-14"
      >
        <h4
          class="MuiTypography-root MuiTypography-h4"
        >
          Please Select Your First Major
        </h4>
        <div
          class="MuiBox-root MuiBox-root-15"
        >
           
          <div
            class="sc-hKgJUU bfONkU inner card-center"
            style="height: 100%;"
          >
            <div
              class="MuiPaper-root MuiCard-root cardselection clickable MuiPaper-elevation1 MuiPaper-rounded"
            >
              <div
                class="MuiCardHeader-root capit"
              >
                <div
                  class="MuiCardHeader-content"
                >
                  <span
                    class="MuiTypography-root MuiCardHeader-title MuiTypography-h5 MuiTypography-displayBlock"
                  >
                    string
                  </span>
                  <span
                    class="MuiTypography-root MuiCardHeader-subheader MuiTypography-body1 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
                  >
                    Major
                  </span>
                </div>
              </div>
              <img
                alt="Major Card"
                class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img"
                height="194"
                src="https://images.squarespace-cdn.com/content/v1/5e408d9d0b0aed2f2768b9aa/1585703579456-3UVMWL3XQM0XIIUHHQS8/faq-temp-img.png"
              />
              <div
                class="MuiCardContent-root"
              >
                <p
                  class="MuiTypography-root maxHeight MuiTypography-body1"
                >
                  string
                </p>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiBox-root MuiBox-root-16"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              Select
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="MuiBox-root MuiBox-root-17"
        >
          <div
            class="MuiBox-root MuiBox-root-18"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root navigationIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
              />
            </svg>
          </div>
          <div
            class="MuiBox-root MuiBox-root-19 navIndicator activewoopog"
          />
          <div
            class="MuiBox-root MuiBox-root-20"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root navigationIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Units First works when Renders and matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiBox-root MuiBox-root-3 sc-jrAFXE gWuEVR"
      >
         
        <div
          class="MuiBox-root MuiBox-root-4"
        >
          <span
            class="css-kt6w9b"
          >
            <span
              class="css-102m2hb"
            />
            <span
              class="css-1hljjer"
            />
          </span>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiBox-root MuiBox-root-3 sc-jrAFXE gWuEVR"
    >
       
      <div
        class="MuiBox-root MuiBox-root-4"
      >
        <span
          class="css-kt6w9b"
        >
          <span
            class="css-102m2hb"
          />
          <span
            class="css-1hljjer"
          />
        </span>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
